packages/sdk/src/mint/mint.ts

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { GAS, ONE_YOCTO, TOKEN_METHOD_NAMES } from '../constants';
+import { NearContractCall } from '../execute';
+
+
+export type MintArgs =  {
+  nftContractId: string;
+  reference: string;
+  ownerId: string;
+  options?: MintOptions;
+};
+
+export type MintOptions = {
+    splits?: Splits;
+    amount?: number;
+    royaltyPercentage?: number;
+}
+
+export type Splits = Record<string, number>;
+
+/**
+ * Mint a token given via reference json on a given contract with a specified owner, amount of copies as well and royalties can be specified via options
+ * @param mintArguments {@link MintArgs}
+ * @returns contract call to be passed to @mintbase-js/sdk execute method
+ */
+export const mint = (
+  args: MintArgs,
+): NearContractCall => {
+  const { nftContractId, reference, ownerId, options = {}  } = args;
+
+  const { splits, amount, royaltyPercentage } = options;
+  //0.5 -> 5000
+  adjustSplitsForContract(splits);
+
+  if (splits && Object.keys(splits).length > 50) {
+    console.error('Splits cannnot have more than 50 entries');
```

Comment:
Should we be throwing these errors vs just logging them?  In cases where minting is going to fail for the end user I would think we wan to stop bad args from making it to NEAR at all